experimental/plugin.go

```diff
@@ -0,0 +1,30 @@
+package experimental
+
+import (
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+// PluginExe is the singleton container for your plugin.
+type PluginExe interface {
+	// CheckHostHealth for the plugin executable
+	CheckExeHealth(config backend.PluginConfig) *backend.CheckHealthResult
+
+	// request for a new datasource
+	NewDataSourceInstance(config backend.PluginConfig) (DataSourceInstance, error)
+}
+
+// DataSourceInstance implements each of the supported requests.  Alternativly this could be a set
+// of interfaces that work for DataSource|AlertNotifier|etc|etc... with the support interrogated and
+// maintaind by the helper on startup
+type DataSourceInstance interface {
```

Comment:
Generally don't want to export interfaces from an API generally.

Reply:
what do you export instead?

Reply:
Due to https://github.com/golang/tools/tree/master/internal/apidiff#interfaces , *if I understand it correctly* Better to have a struct and include individual interfaces like in http://localhost:6060/pkg/github.com/grafana/grafana-plugin-sdk-go/backend/#ServeOpts

Reply:
discussed... and we can improve later if needed